python/nav/ipdevpoll/plugins/paloaltoarp.py

```diff
@@ -0,0 +1,160 @@
+#
+# Copyright (C) 2023 University of Tromsø
+#
+# This file is part of Network Administration Visualized (NAV).
+#
+# NAV is free software: you can redistribute it and/or modify it under the
+# terms of the GNU General Public License version 3 as published by the Free
+# Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+# more details.  You should have received a copy of the GNU General Public
+# License along with NAV. If not, see <http://www.gnu.org/licenses/>.
+#
+
+"""
+    ipdevpoll plugin for fetching arp mappings from Palo Alto firewalls
+
+    Add [paloaltoarp] section to ipdevpoll.conf
+    add hostname = key to [paloaltoarp] section
+    for example:
+    [paloaltoarp]
+    10.0.0.0 = abcdefghijklmnopqrstuvwxyz1234567890
+
+"""
+
+from twisted.internet import defer, reactor, ssl
+from twisted.internet.defer import returnValue
+from twisted.web.client import Agent
+from twisted.web import client
+from twisted.web.http_headers import Headers
+
+from nav.ipdevpoll.plugins.arp import Arp
+from nav.config import getconfig
+
+from nav import buildconf
+
+import xml.etree.ElementTree as ET
+from IPy import IP
+
+
+def parse_arp(arp):
+    """
+    Create mappings from arp table
+    xml.etree.ElementTree is considered insecure: https://docs.python.org/3/library/xml.html#xml-vulnerabilities
+    However, since we are not parsing untrusted data, this should not be a problem.
+    """
+
+    arps = []
+
+    root = ET.fromstring(arp)
+    entries = root[0][4]
+    for entry in entries:
+        status = entry[0].text
+        ip = entry[1].text
+        mac = entry[2].text
+        if status != " i ":
+            if mac != "(incomplete)":
+                arps.append(('ifindex', IP(ip), mac))
+
+    return arps
+
+
+class PaloaltoArp(Arp):
+    def __init__(self, *args, **kwargs):
+        self._logger.debug("PaloaltoArp initialized")
+
+        self.paloalto_devices = []
+        ipdevpoll_config = getconfig('ipdevpoll.conf')
```

Comment:
This bit is actually unnecessary.  The ipdevpoll configuration is already made available to the plugin, you can find it in `self.config` :)